app/routes/admin/permissions/system-roles.js

```diff
@@ -5,5 +5,16 @@ const { Route } = Ember;
 export default Route.extend({
   titleToken() {
     return this.l10n.t('System Roles');
+  },
+  model() {
+    return this.get('store').findAll('user-permission');
+  },
+  actions: {
+    willTransition() {
+      let model = this.get('controller.model');
+      model.forEach(function(userPermission) {
```

Comment:
Why store it in a variable ? Do a foreach directly on `this.get('controller.model')`